app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -260,12 +274,27 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
 
     private fun observeReceivedTabs(): Disposable {
         return serviceLocator!!.fxaRepo.receivedTabs.subscribe {
-            // TODO: Gracefully handle receiving multiple tabs around the same time. #2777
-            serviceLocator!!.screenController.showBrowserScreenForUrl(fragmentManager!!, it.url)
-            ViewUtils.showCenteredBottomToast(context, it.sourceDescription.resolve(resources))
+            val appIsInForeground = lifecycle.currentState == Lifecycle.State.STARTED ||
+                lifecycle.currentState == Lifecycle.State.RESUMED
+            val appMightBeInBackground = lifecycle.currentState == Lifecycle.State.CREATED
+
+            when {
+                appIsInForeground -> receiveTab(it)
+                appMightBeInBackground -> {
+                    serviceLocator!!.fxaRepo.lastTabCouldNotBeDisplayed()
+                    context?.application?.bringAppToForeground()
```

Comment:
afaict, opening the app is not part of the acceptance criteria:

> If tab is received while Firefox is in background, show received tab **when Firefox is opened**

> User sees their latest received tab **the next time Firefox is opened** (the currently open tab is added to backstack, as they would be if Firefox was open)

I wonder if not handling this will make it simpler to handle the Fragment transaction exception.

Reply:
I checked with Bram and this was his expectation of the behavior.  I think this might be jumping ahead a bit, but it's definitely functionality we need.  AFAICT it wouldn't help the transaction problem, though (that's really just not knowing when to first navigate to the tab after startup).

TBH I didn't notice it wasn't part of the A-C, and over-implemented on accident.

Reply:
If we're waiting on a follow up, I'd rather leave the code as is and change if necessary, in order to avoid unnecessary work.